UsageStatistics_StorageReader.php

```diff
@@ -32,35 +57,34 @@ public function get_history_summary() {
 			$summary = array( 'period' => array() );
 		} else {
 			$timestamp_start = $history[0]['timestamp_start'];
-			$timestamp_end = $history[count( $history ) - 1]['timestamp_end'];
+			$timestamp_end   = $history[ count( $history ) - 1 ]['timestamp_end'];
 
 			$period = array(
-				'timestamp_start' => $timestamp_start,
-				'timestamp_start_mins' =>
-				Util_UsageStatistics::time_mins( $timestamp_start ),
-				'timestamp_end' => $timestamp_end,
-				'timestamp_end_mins' =>
-				Util_UsageStatistics::time_mins( $timestamp_end ),
+				'timestamp_start'      => $timestamp_start,
+				'timestamp_start_mins' => Util_UsageStatistics::time_mins( $timestamp_start ),
+				'timestamp_end'        => $timestamp_end,
+				'timestamp_end_mins'   => Util_UsageStatistics::time_mins( $timestamp_end ),
 			);
 
-			$period['seconds'] = $timestamp_end - $timestamp_start;
-			$summary['period'] = $period;
+			$period['seconds']       = $timestamp_end - $timestamp_start;
+			$summary['period']       = $period;
 			$summary['timeout_time'] = time() + 15;
 
-			$summary = apply_filters( 'w3tc_usage_statistics_summary_from_history',
-				$summary, $history );
+			$summary = apply_filters( 'w3tc_usage_statistics_summary_from_history', $summary, $history );
 		}
 
-		$summary['period']['to_update_secs'] = (int)$w->get_hotspot_end() - time() + 1;
+		$summary['period']['to_update_secs'] = (int) $w->get_hotspot_end() - time() + 1;
 
 		unset( $summary['memcached_servers'] );
 		unset( $summary['redis_servers'] );
 
-		while ( count($history) < $this->items_to_return ) {
+		$history_count = count( $history );
+		while ( $history_count < $this->items_to_return ) {
```

Comment:
This will cause an indefinite loop.  The `while` condition must check the `count` of `$history` for every iteration.  You can change it to `while ( true )` and `break` if the condition is met.

Reply:
Fixed